### ISSUE TITLE
Intercept search bar string?

### ISSUE THREAD

**Issue by mgrauer** [girder/girder_web_components#164]
Should the search bar in the hierarchy widget be intercept-able and extensible so that e.g. a custom DSL could be used, and then sent on to a custom end point?

This might generalize into a question of: should the search bar support various search modes, which allow for different strings and different endpoints, potentially affecting the search results display?

Should this be a change in the library or project specific?

**Comment by matthewma7** [comment]
I could be wrong, I think the current purpose of the [search component](https://github.com/girder/girder_web_components/blob/master/src/components/Search.vue) (screenshot below)  is 'girder search component', it's probably not designed to be open up for customization too much. 
So, start something new with Vuetify's [Combobox](https://vuetifyjs.com/en/components/combobox) probably is a better choice considering all the customization mentioned above. 

![image](https://user-images.githubusercontent.com/3123478/66125427-e461cf00-e5b4-11e9-8328-90bb95bd072d.png)

**Comment by subdavis** [comment]
I agree that I don't want to allow for much customization on the actual display of the search widget.  

Mike is asking about something like this (or like the github issue/PR search feature):

![Screenshot from 2019-10-03 09-02-15](https://user-images.githubusercontent.com/4214172/66128788-96e96000-e5bc-11e9-962d-9cb56f72d930.png)

The only change I'm aware of needed to support something like that is extension of the `searchResults` computed property to parse the searchText and fire a query at a different endpoint.

Whether or not that's appropriate for this library is definitely worth debate.

**Comment by zachmullen** [comment]
Girder actually supports pluggable "search modes" for the single-string search REST API, so the normative Girder way to do this would render front-end pluggability unnecessary.

**Comment by matthewma7** [comment]
If we would want to just intercept search terms and call a new endpoint, it can be achieved by monkey patching, which I have done to the search component before. But if we also would like to customize the display of the result such as adding a thumbnail. I would lean towards recreating some new, even start from copy and paste the search component isn't too bad since there is only 200 line of code. 

I have a similar opinion for https://github.com/girder/girder_web_components/issues/165. So allow me to link to it here.

**Comment by subdavis** [comment]
Adding a scoped slot to the search results to populate the contents of a v-list-tile seems like a reasonable customization to me.  I know we're constantly walking the line between feature creep and practicality, but thus far I've found the search bar too rigid to be useful.   Writing it from scratch feels like a missed opportunity to find the sweet spot of usefulness.

I'd also like to avoid recommending monkey patching to library consumers as anything other than a last resort.

**Comment by matthewma7** [comment]
> Writing it from scratch feels like a missed opportunity to find the sweet spot of usefulness
Great point.

**Comment by subdavis** [comment]
Per our discussion, this will be done by adding support for additional search modes, and search results will be provided by a girder plugin.  This should be nearly transparent to GWC.  Just need to add props for additional search modes and the default search mode in search widget.

**Comment by zachmullen** [comment]
More specifically, the following customizability features will be desired for DANDI:

* Rather than the search mode and search types being exclusively internal data of the search component, they should both be settable via props.
* Internal search mode and search types should be able to be disabled by a prop switch, so that the right-hand menu doesn't show up at all.
